public/components/Callback.js

```diff
@@ -36,9 +40,16 @@ function Callback() {
       } else {
         getData(`${process.env.API_URL}/api/v1.0/auth/permissions`, token)
           .then((data) => {
-            putPermissions(data);
-            setInfoMessage("You're logged in.");
-            window.location.replace("/");
+            if (data.length > 0) {
+              putPermissions(data);
+              setInfoMessage("You're logged in.");
+              window.location.replace("/");
+            } else {
+              putPermissions(data);
```

Comment:
should we `putPermissions` here, when data has no length? Or is it setting `[]`?

Reply:
It's setting []. I also thought about not setting it, but I think it's better we save it empty so we know we don't have to request it anymore. If you have a different logic on it, let me know.